packages/alfa-act/src/interview.ts

```diff
@@ -5,9 +5,26 @@ import { Oracle } from "./oracle";
 import { Question } from "./question";
 import { Rule } from "./rule";
 
-export type Interview<Q, S, T> =
+/**
+ * As `Interview` is a recursive type that models nested chains of `Question`s,
+ * we need to limit the depth to which this recursion can happen to avoid the
+ * TypeScript compiler going into an infinite type resolution loop. This is done
+ * by maintaining a pointer, `D extends number` that down to -1 at which point
+ * the recursion will stop. Given a pointer `D`, the `Depths` tuple provides
+ * `D - 1 = Depths[D]`.
+ */
+type Depths = [-1, 0, 1, 2];
```

Comment:
For the record, this is an ugly trick 🤯 

Also, this means we now have a maximum number of nested questions we can have. Not a huge deal and can easily be augmented anyway…

Reply:
The wonders of Turing completeness 🙈 Yes, it's an unfortunate but necessary trick that I snatched straight from the TypeScript compiler itself:

https://github.com/microsoft/TypeScript/blob/4bda7ce8e58e3e6393aff0f9bc23956e0eeb8a4e/lib/lib.es2019.array.d.ts#L21-L26

If we end up in a situation where we need to nest more than 4 levels of questions, we'll adjust the `Depths` type accordingly. 4 levels is the current need.